pkg/operation/botanist/waiter.go

```diff
@@ -70,7 +71,7 @@ func (b *Botanist) WaitUntilEtcdReady(ctx context.Context) error {
 
 		bothEtcdStatefulSetsReady := true
 		for _, statefulSet := range statefulSetList.Items {
-			if statefulSet.DeletionTimestamp != nil {
+			if statefulSet.DeletionTimestamp != nil || statefulSet.Generation != statefulSet.Status.ObservedGeneration {
```

Comment:
Why was this added? If the generation is not equals to the observed generation then we should set `bothEtcdStatefulSetsReady=false`.

Reply:
Ahh yeah . Not needed.